Lib/inspect.py

```diff
@@ -285,7 +286,24 @@ def isroutine(object):
 
 def isabstract(object):
     """Return true if the object is an abstract base class (ABC)."""
-    return bool(isinstance(object, type) and object.__flags__ & TPFLAGS_IS_ABSTRACT)
+    if isinstance(object, type):
+        if object.__flags__ & TPFLAGS_IS_ABSTRACT:
+            return True
+        elif (issubclass(type(object), abc.ABCMeta)
+                and not hasattr(object, '__abstractmethods__')):
+            # We're likely in the __init_subclass__ of an ABC. ABCMeta.__new__
+            # hasn't finished running yet. We have to search for abstractmethods
+            # by hand. Code copied from ABCMeta.__new__.
+            abstracts = {name
```

Comment:
It seems to me that collecting names of abstract method is not needed. 
`False` can be returned just after finding any abstract method.